actors.md

```diff
@@ -1132,6 +1059,33 @@ func IsSlashed() (bool) {
 }
 ```
 
+#### `AddFaults`
+
+**Parameters**
+
+```sh
+type AddFaults struct {
+    faults FaultSet
+} representation tuple
+```
+
+**Algorithm**
+
+```go
+func AddFaults(faults FaultSet) {
```

Comment:
I actually think we should simplify this all the way to just:
```
func AddFaults(faults FaultSet) {
  self.currentFaultSet = Merge(self.currentFaultSet, faults)
}
```

If you submit faults before you submit your post, they apply to the post you still havent submitted.

Reply:
If we simplify this, it means you are not allowed to submit any faults between challenge time and post submission. Which becomes an issue if you miss a post, as you can't post new faults anymore.

Reply:
This would allow a miner to accept storage deals, commit and then discard sectors containing the storage, and then add faults each proving period thereafter. Since the fault set uses RLE, the cost of a the AddFault message is constant over a large number of sectors, and the cost per sector drops without bound. At some point the gas costs for faulting on all deals will drop below the value of the deals.

We're going to need some mechanism to limit the number of times a sector can be faulted or at least a mechanism for timing-out faulted sectors so deal arbitration and payment cancellation can occur.